### PR TITLE
rodeos: Add support for stream_wapper_v1 with std::string route 📦 

### PR DESCRIPTION
**Pull request by heifner**
## Change Description

- Add new `stream_wrapper_v1` with `std::string` route which removes the restriction to use `eosio::name` for routing keys.

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [ ] Stability bug fix

- [x] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: programs/rodeos/streams/logger.hpp [hunk 1]
```diff
@@ -6,22 +6,22 @@
...
+   explicit logger(std::vector<std::string> routes)
+   : stream_handler(std::move(routes)) {
+      ilog("logger initialized");
+   }
 
-   void publish(const std::vector<char>& data, const eosio::name& routing_key) override {
-      ilog("logger stream [${data_size}] >> ${data}", ("data", data)("data_size", data.size()));
+   void publish(const std::vector<char>& data, const std::string& routing_key) override {
+      ilog("logger stream ${r}: [${data_size}] >> ${data}",
+           ("r", routing_key)("data", data)("data_size", data.size()));
```
Comment (linhuang-blockone) [inline programs/rodeos/streams/logger.hpp:17]: The order of "data" and "data_size" is reversed.

Comment (heifner) [reply programs/rodeos/streams/logger.hpp:17]: The order doesn't matter.

**Review by linhuang-blockone — COMMENTED** [review]
Just curious why the type of routes is changed from eosio::name to std::string?

**Review by linhuang-blockone — APPROVED** [review]

**Comment by heifner** [comment]
> Just curious why the type of routes is changed from eosio::name to std::string?

eosio::name is limited to 12 characters but routing keys in rabbitmq has no such restriction. This allows users to use much more reasonable sized routing keys.

**Pull request merged**